store/tikv/2pc.go

```diff
@@ -323,6 +325,12 @@ func (c *twoPhaseCommitter) doActionOnKeys(bo *Backoffer, action twoPhaseCommitA
 	var batches []batchKeys
 	var sizeFunc = c.keySize
 	if action == actionPrewrite {
+		if c.regionTxnSize == nil {
+			c.regionTxnSize = make(map[RegionVerID]int)
+		}
+		for region, keys := range groups {
+			c.regionTxnSize[region] = len(keys)
```

Comment:
On region errors, txn size will become the batch size instead of 0 because the failed batch will go through the whole prewrite process. Then, it will be as bad as the original code when we have large value size :(

Reply:
There will be some other problems when we use `region.ID` as the map key. When there are several batches in a new region retrying concurrently, it is a problem how we update the txn size of the region...

Reply:
oh good point :rofl: